Tests/TuistGeneratorTests/Generator/ProjectGroupsTests.swift

```diff
@@ -104,31 +105,31 @@ final class ProjectGroupsTests: XCTestCase {
                                    targets: [target1, target2, target3, target4])
 
         // When
-        subject = ProjectGroups.generate(project: project,
-                                         pbxproj: pbxproj,
-                                         xcodeprojPath: project.path.appending(component: "\(project.fileName).xcodeproj"),
-                                         sourceRootPath: sourceRootPath,
-                                         playgrounds: playgrounds)
+        subject = ProjectGroups.generateInitialGroups(project: project,
+                                                      pbxproj: pbxproj,
+                                                      xcodeprojPath: project.path.appending(component: "\(project.fileName).xcodeproj"),
+                                                      sourceRootPath: sourceRootPath,
+                                                      playgrounds: playgrounds)
 
         // Then
-        let paths = subject.main.children.compactMap { $0.nameOrPath }
+        let paths = subject.sortMainAndAddDefaultGroups().children.compactMap { $0.nameOrPath }
         XCTAssertEqual(paths, [
-            "P",
+            "A",
             "B",
             "C",
-            "A",
+            "P",
             "Frameworks",
             "Playgrounds",
             "Products",
         ])
```

Comment:
ouch we regressed this - we currently respect the order of custom groups names defined by targets in the order they are defined.

We could potentially drop this as the main use case was when we used to create a `Manifest` target and we always wanted the `Manifest` target files to appear below `Project` - seeing we no longer add the manifest target in generated projects there is no need to support this. If for whatever reason we need to - we can revisit this :)

Reply:
I've now changed the sort now to not the root level folders in the tree and only sort their children.